File: musical-robot/src/musical-robot/Models/MissingPerson.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace musical_robot.Models
{
	public class MissingPerson
	{
		public string UID { get; set; }
		public string Surname { get; set; }
		public string Forename { get; set; }
		public string Gender { get; set; }
		public int? BirthYear { get; set; }
		public int? Age
		{
			get
			{
				if (BirthYear.HasValue)
				{
					return DateTime.Now.Year - BirthYear.Value;
				}
				return null;
			}
		}
		public string Status { get; set; }
		public string StatusPriorToDormant { get; set; }
		public string Category { get; set; }
		public string AccomodationType { get; set; }
		public string Borough { get; set; }
		public string OutputArea { get; set; }
		public double Latitude { get; set; }
		public double Longitude { get; set; }
		public string OutputAreaCenX { get; set; }
		public string OutputAreaCenY { get; set; }
		public string MissingOn { get; set; }
		public string RecordCreatedOn { get; set; }
		public string RecordUpdatedOn { get; set; }
		public string LastSeenOn { get; set; }
		public string StatusChangedOn { get; set; }
		public IEnumerable<History> History { get; set; }
	}

	public class History
	{
		public string Message { get; set; }
		public DateTime EntryOn { get; set; }
		public Coords Location { get; set; }
	}

	public class Coords
	{
		public double Latitude { get; set; }
		public double Longitude { get; set; }
	}
}

File: musical-robot/src/musical-robot/app/missingperson.ts
import { History } from './history';

export class MissingPerson {
	constructor(
		public uid: string,
		public surname: string,
		public forename: string,
		public birthYear: number,
		public age: number,
		public status: string,
		public statusPriorToDormant: string,
		public category: string,
		public accomodationType: string,
		public borough: string,
		public outputArea: string,
		public latitude: number,
		public longitude: number,
		public outputAreaCenX: string,
		public outputAreaCenY: string,
		public missingOn: string,
		public recordCreatedOn: string,
		public recordUpdatedOn: string,
		public lastSeenOn: string,
		public statusChangedOn: string,
		public history: History) { }
}